oversweet-api/src/main/java/com/depromeet/oversweet/member/controller/MemberController.java

```diff
@@ -0,0 +1,73 @@
+import com.depromeet.oversweet.member.service.MemberFacade;
+import com.depromeet.oversweet.response.DataResponse;
+import com.depromeet.oversweet.util.JwtTokenProvider;
+import com.depromeet.oversweet.util.TokenResponse;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@Slf4j
+@Tag(name = "회원", description = "회원 관련 API")
+@RestController
+@RequestMapping("/api/v1/members")
+@RequiredArgsConstructor
+public class MemberController {
+
+    private final MemberFacade memberFacade;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    @Operation(summary = "닉네임 중복 체크", description = "닉네임 중복 체크 API 입니다.")
+    @ApiResponses({@ApiResponse(responseCode = "200", description = "닉네임 중복 체크 성공")})
+    @PostMapping("/validation")
+    public ResponseEntity<DataResponse<NicknameValidationResponse>> checkNicknameDuplicated(@RequestParam String nickname) {
+        NicknameValidationResponse validationResponse = memberFacade.checkNicknameDuplicated(nickname);
+        return ResponseEntity.ok().body(DataResponse.of(HttpStatus.OK, "닉네임 중복 체크 성공", validationResponse));
+    }
+
+    @Operation(summary = "카카오 회원가입 및 로그인 성공", description = "카카오 로그인 API 입니다.")
+    @ApiResponses({@ApiResponse(responseCode = "200", description = "카카오 회원가입 및 로그인 성공")})
+    @PostMapping("/kakao")
+    public ResponseEntity<DataResponse<SocialSignInResponse>> kakaoSignIn(@RequestBody final KakaoTokenRequest kakaoTokenRequest,
+                                                                          HttpServletResponse httpServletResponse) {
+        SocialSignInResponse socialSignInResponse = memberFacade.signInKakao(kakaoTokenRequest);
+
+        if (!socialSignInResponse.isNewMember()) {
+            TokenResponse tokenResponse = jwtTokenProvider.generateJwtToken(socialSignInResponse.id());
+            httpServletResponse.setHeader(AUTHORIZATION, tokenResponse.accessToken());
+            httpServletResponse.addCookie(new Cookie("refreshToken", tokenResponse.refreshToken()));
+        }
+
+        return ResponseEntity.ok().body(DataResponse.of(HttpStatus.OK, "카카오 회원가입 및 로그인 성공", socialSignInResponse));
```

Comment:
HttpServletResponse을 사용하는 것 보단, HttpHeader 객체를 만들어서 값을 주입 해주고, ResponseEntity에 넣어주는 방향은 어떤가요?
```suggestion
        
    HttpHeaders headers = new HttpHeaders();
    CookieUtil.setRefreshCookie(headers, responseDto.getRefreshToken());
    HttpHeaderUtil.setAccessToken(headers, responseDto.getAccessToken());

    return new ResponseEntity<>(DataResponse.of(HttpStatus.CREATED,
        "카카오 회원가입 및 로그인 성공, socialSignInResponse), headers, HttpStatus.CREATED);
```

Reply:
아 이 부분을 제가 아까 놓쳤다고 생각한건데,

그 기존의 경우 이 api 를 통해 바로 로그인을 해서 홈으로 진입을 할거고
신규면 다시 추가 정보 기입하는 api 를 만나게 될거라서 분기 처리를 했습니다. 그런데 생각해보니 추가 정보 기입 후 회원가입 하는 api 에 token 을 추가하지 않았네요..

즉,

> 기존인데 로그인이 풀려서 다시 로그인을 하게 될 경우 여기서 token 을 얻는다.
> 신규라서 여기서는 카카오 로그인만 하고 다시 추가 정보 기입하는 화면을 만나서 기입 후 회원가입을 하면 그 api 에서 token 을 얻는다.

이렇게 되어야 할 것 같아요! 

(그 위에 작성해주신 내용은 다시 코멘트 작성하겠습니다.)

Reply:
저도 처음에는 신규/기존을 구분하지 않고 토큰을 반환한 뒤 _필수 정보 기입했냐_ 라는 boolean 값으로 구분할 수 있을거라 생각해서 신호님께 말씀드려보니, 신호님께서는 이렇게 답변을 주셨어요. 

> _필수 정보 기입했냐_ 라는 boolean 값으로만 분기처리해도 될 거 같음.
> 다만 실무에서는 이렇게 할 경우 히스토리 관리가 잘 안 될 거 같고 어떻게든 다른 사람이 정현님 코드의 의도를 이해하지 못 할 거 같다는 생각이 든다.

그래서 고민 끝에 정말 기존 회원이 로그인하려고 접근했을 때만 토큰을 주자라고 판단했어요.

Reply:
그리고 또 생각해봤는데, 추가 정보 기입 전에 탈주했다가 다시 돌아와서 기입을 했는데 이때 토큰을 다시 발행해주는 로직이 없다면 어떤 유저는 회원가입했는데 토큰이 만료되거나 하는 이슈도 있을 것 같아서

신규/기존 구분 없이 토큰을 발행해주더라도
필수 정보를 기입하고 가입하는 곳에서 다시 토큰을 발행해주는 로직이 필요할 것 같긴 합니다.

Reply:
신호님 의견도 맞는 말씀인 것 같아요. 결국 의도를 어떻게 들어내냐도 중요한 점이기 때문이에요. 결국 토큰을 발행하는 시점이 문제인 것 같은데, 이 부분은 유저가 처음 카카오 로그인을 완료해서 save가 호출 됐을 때, 필수 정보를 기입하는 공간에서 다시 들어왔을 때 주면 될 것 같아요.

1. 카카오 로그인 성공 -> 회원가입/로그인 완료 -> 토큰 발급
2. 이탈 후 재접속. -> 해당 유저가 이미 가입한 적이 있는지 (카카오 소셜 Id로 검증) -> 있다면 토큰 발급

이렇게 갈 수 있을 것 같아요.

Reply:
넵 제준님

즉, 정리하면

> 카카오 로그인 했는데 필수 정보까지 모두 다 기입한 기존 유저의 경우 -> 토큰 발행
> 카카오 로그인 했는데 처음 가입하려고 진입한 유저의 경우 -> 토큰 발행 X & 추가 정보 기입 화면으로 이동시키기
> 추가 정보 기입한 뒤 우리 서비스에 완전히 가입한 유저의 경우 -> 토큰 발행

이렇게 정리가 될 것 같아요!

이 부분 확정되면 수정 진행할게요~

Reply:
일단 로그인 관련 api 가 2개입니다.(애플 포함하면 3개)

- kakao : 카카오 로그인 시작하기 버튼을 누른 뒤 클라에서 처리하고 나서 호출하는 api => 편의상 카카오 api
- auth : 기입한 추가 정보를 저장하는 api => 편의상 추가 정보 api

즉, 카카오 로그인을 시도했는데(위 api 1번 상황)

***1번 API***

- 기존 멤버 데이터도 있고 && 필수 정보도 기입했다면
    - **토큰 발행 O**
- 신규 & 필수 정보 X
    - **토큰 발행 X**
    - 2번 API 이동
    

***2번 API***

- 추가 정보 저장
- **토큰 발행 O**